bundles/core/src/main/java/com/adobe/cq/commerce/demandware/InstanceIdProvider.java

```diff
@@ -0,0 +1,34 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2019 Adobe Systems Incorporated
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+
+package com.adobe.cq.commerce.demandware;
+
+import com.day.cq.replication.AgentConfig;
+import com.day.cq.wcm.api.Page;
+import org.apache.sling.api.SlingHttpServletRequest;
+
+/**
+ * Component demandware instance id service to identify the instance id of configured demandware.
+ */
+public interface InstanceIdProvider {
+    
+    String getInstanceId(Page page);
+    
+    String getInstanceId(SlingHttpServletRequest request);
```

Comment:
the instanceID should not be derived from the referrer. If we need the ID for subsequent requests... maybe it can be transmitted explicitly via selector or request parameter? what is the use case?